server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -51,13 +58,18 @@ void recordFileTransferStartTime(long uploadStartTime) {
         }
     }
 
-    void recordBytesForFiles(Set<TransferFileSnapshot> toUpload) {
+    void recordBytesForFiles(Set<TranslogCheckpointSnapshot> toUpload) {
         bytesForTlogCkpFileToUpload = new HashMap<>();
         toUpload.forEach(file -> {
             try {
-                bytesForTlogCkpFileToUpload.put(file.getName(), file.getContentLength());
+                bytesForTlogCkpFileToUpload.put(file.getTranslogFileName(), file.getTranslogFileContentLength());
             } catch (IOException ignored) {
-                bytesForTlogCkpFileToUpload.put(file.getName(), 0L);
+                bytesForTlogCkpFileToUpload.put(file.getTranslogFileName(), 0L);
+            }
+            try {
+                bytesForTlogCkpFileToUpload.put(file.getCheckpointFileName(), file.getCheckpointFileContentLength());
+            } catch (IOException ignored) {
+                bytesForTlogCkpFileToUpload.put(file.getCheckpointFileName(), 0L);
```

Comment:
This code is duplicated from just the above try catch block. Can we make a private method and just use them across both.

Reply:
In case of w/o metadata, should we exclude the files that have been uploaded already?

Reply:
make sense.